lib/cache.gd

```diff
@@ -30,23 +30,44 @@
 ##  or GAP chooses to flush all caches, or if multiple threads try to calculate
 ##  the same value simultaneously.
 ##  <P/>
-##  <A>defaults</A>, if given, is used to initalise the cache.
+##  The optional second argument is a record which provides a number
+##  of configuration options. The following options are supported.
+##  <List>
+##  <Mark><C>defaults</C> (default an empty list)</Mark>
+##  <Item>
+##    Used to initalise the cache, both initially and after each flush.
+##  </Item>
+##  <Mark><C>flush</C> (default <K>true</K>)</Mark>
+##  <Item>
+##    If this is <K>true</K>, the cache is emptied whenever
+##    <Ref Func="FlushCaches"/> is called.
+##  </Item>
+##  <Mark><C>errorHandler</C> (defaults to <Ref Func="Error"/>)</Mark>
+##  <Item>
+##    A function to be called when a non-positive integer is given to
+##    the cache. If this function returns a value, that value is returned.
```

Comment:
Maybe say "... when an input that is not a positive integer is given ..." (that could also be a bool, string, ...)